app/src/main/java/com/datastax/themis/cli/commands/InsertCommand.java

```diff
@@ -44,56 +35,50 @@ public class InsertCommand implements Callable<Integer> {
 
     private final ImmutableMap<ClusterName, Cluster> clusters;
 
-    /* Safe to re-use only because we're resetting values on the same columns each time and not alterting anything more substantive */
-    private final InsertInto insertBuilder =
-            QueryBuilder.insertInto(Constants.DEFAULT_KEYSPACE, Constants.DEFAULT_TABLE);
-
-    private final Statement maxQueryStmt =
-            QueryBuilder.selectFrom(Constants.DEFAULT_KEYSPACE, Constants.DEFAULT_TABLE)
-            .function("max", Selector.column("key"))
-                    .where(Relation.column("app").isEqualTo(QueryBuilder.literal("themis")))
-            .build();
-
     public InsertCommand(ImmutableMap<ClusterName, Cluster> clusters) {
         this.clusters = clusters;
     }
 
     @Override
     public Integer call() throws Exception {
 
+        /* Safe to re-use only because we're resetting values on the same columns each time and not altering anything more substantive */
+        InsertInto insertBuilder =
+                QueryBuilder.insertInto(this.keyspace, this.table);
+
+        Statement maxQueryStmt =
+                QueryBuilder.selectFrom(this.keyspace, this.table)
+                        .function("max", Selector.column("key"))
+                        .where(Relation.column("app").isEqualTo(QueryBuilder.literal("themis")))
+                        .build();
+
+        /* My kingdom for real partial application */
+        Function<ClusterName, Boolean> insertFn = (ClusterName clusterName) -> {
+            return insertIntoCluster(clusterName, insertBuilder, maxQueryStmt);
+        };
```

Comment:
Here (and in the other commands) we now rely on params that aren't available when this instance is constructed; presumably they're added after the fact via reflection.  As a result we can no longer pre-generate these statements and builders statically, but we can at least do it once in the implementation of call().  This _does_ mean that we need to pass all the state we rely on into the function call, though, so to facilitate that without tons of duplication I leveraged this function definition as an astonishingly crude version of partial application.  There's still auto-boxing and unboxing going on here... but at least we avoid the code cruft. :thisisfine: